asllib/doc/SubprogramCalls.tex

```diff
@@ -112,150 +91,198 @@ \subsubsection{Formally}
+  \item applying $\unziplist$ to $\texttt{typed\_args}$ yields the corresponding lists of types
+        and expressions $\argtypes$ and $\args$;
+  \item applying $\subprogramforname$ to match $\name$ and $\argtypes$ in $\tenv$
+        yields \\ $(\namep, \funcsig)$\ProseOrTypeError;
+  \item either the $\subprogramtype$ of $\funcsig$ equals $\calltype$,
+        or the $\subprogramtype$ of $\funcsig$ is $\STGetter$ and $\calltype$ is $\STFunction$\ProseOrTypeError;
+  \item applying $\insertstdlibparam$ to $\funcsig$, $\params$, and $\argtypes$ yields new parameters $\paramsone$;
+  \item checking that the lengths of $\funcsig.\funcparameters$ and $\paramsone$ are the same yields $\True$\ProseOrTypeError;
+  \item checking that the lengths of $\funcsig.\funcargs$ and $\args$ are the same yields $\True$\ProseOrTypeError;
+  \item applying $\checkparamstypesat$ to $\paramsone$
+        to check that the actual parameters have correct types with respect to $\funcsig.\funcparameters$
         in $\tenv$ yields $\True$\ProseOrTypeError;
-  \item applying $\checkcalleeparams$ to $\calleeparams$ to check they have correct types
-        with respect to $\eqsthree$ in $\tenv$ yields $\True$\ProseOrTypeError;
-  \item applying $\annotateretty$ to $\calltype$ and $\callee.\funcreturntype$
+  \item define $\eqs$ as the association of declared parameter names in $\funcsig.\funcparameters$ with actual parameters $\paramsone$;
+  \item applying $\checkargstypesat$ to $\argtypes$ and $\eqs$
+        to check that the actual arguments have correct types with respect to $\funcsig.\funcargs$
+        in $\tenv$ yields $\True$\ProseOrTypeError;
+  \item applying $\annotateretty$ to $\eqs$, $\calltype$, and $\funcsig.\funcreturntype$
         to check that the two call types match and to substitute actual parameter arguments in the formal return type
         yields $\rettyopt$ \ProseOrTypeError;
-  \item define $\eqs$ as $\eqsthree$.
+  \item define $\vcall$ as the call with name $\namep$, parameters taken from $\paramsone$, arguments $\vargs$, and call type $\funcsig.\funcsubprogramtype$.
 \end{itemize}
 \subsubsection{Formally}
 \begin{mathpar}
 \inferrule{
-  \unziplist(\callerargstyped) = (\callerargtypes, \vargsone)\\
-  {
-    \begin{array}{r}
-    \subprogramforname(\tenv, \name, \callerargtypes) \typearrow \\
-  (\eqsone, \nameone, \callee) \OrTypeError
-    \end{array}
-  }\\
-  \checktrans{\callee.\subprogramtype = \calltype}{\MismatchedReturnValue} \checktransarrow \True\OrTypeError\\\\
-  \equallength(\callee.\funcargs, \vargsone) \typearrow \aritymatch\\
-  \checktrans{\aritymatch}{\CallBadArity} \checktransarrow \True \OrTypeError\\\\
+  \unziplist(\texttt{typed\_args}) = (\argtypes, \args)\\
+    \subprogramforname(\tenv, \name, \argtypes) \typearrow
+  (\nameone, \funcsig) \OrTypeError \\
   {
-  \annotateparameterdefining\left(
+  \hyperlink{def-checktrans}{\textfunc{check}}\left(
```

Comment:
I'm not sure what's happending here. You probably want to use the `checktrans{}{}` macro, which allows you to include the error code.